webrender/src/frame_builder.rs

```diff
@@ -1684,96 +1678,54 @@ impl FrameBuilder {
             .get(&pipeline_id)
             .expect("No display list?")
             .display_list;
-        debug!(
-            "\tclip_bounds {:?}, layer_local_clip {:?}",
-            clip_bounds,
-            packed_layer.local_clip_rect
-        );
 
         if !stacking_context.is_backface_visible && packed_layer.transform.is_backface_visible() {
             return;
         }
 
+        let prim_context = PrimitiveContext::new(
```

Comment:
hmm, so this clip-scroll tree traversal used to be cached in `current_clip_info`, and now it's performed unconditionally. Seems like a perf regression?

Reply:
It *is* a performance regression, but only minor (and temporary). Previously, the cache would get invalidated as soon as the clip ID changed. Now, we do it for every primitive run. But primitive runs get broken when the clip and/or scroll ID changes. In the vast majority of cases, the cause for a primitive run break is when the clip node ID has changed, which would invalidate the cache anyway.